tests/components/insteon/test_config_flow.py

```diff
@@ -594,3 +595,56 @@ async def test_options_override_bad_data(hass: HomeAssistant):
 
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
     assert result["errors"] == {"base": "input_error"}
+
+
+async def test_discovery_via_usb(hass):
+    """Test usb flow."""
+    discovery_info = usb.UsbServiceInfo(
+        device="/dev/ttyINSTEON",
+        pid="AAAA",
+        vid="AAAA",
+        serial_number="1234",
+        description="insteon radio",
+        manufacturer="test",
+    )
+    result = await hass.config_entries.flow.async_init(
+        "insteon", context={"source": config_entries.SOURCE_USB}, data=discovery_info
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "confirm_usb"
+
+    with patch(
+        "homeassistant.components.insteon.config_flow._async_connect", return_value=True
```

Comment:
We want to patch the library not our code.
```suggestion
        "homeassistant.components.insteon.config_flow.async_connect"
```